python/GafferSceneTest/DeletePointsTest.py

```diff
@@ -149,6 +149,100 @@ def testBoundsUpdate( self ) :
+		testObject["outOfRangeIds"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Vertex, IECore.Int64VectorData( [ i + 8000000000 for i in range( 10 ) ] ) )
+		testObject["duplicateIds"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Vertex, IECore.Int64VectorData( [ 0, 0, 1, 1, 2, 2, 3, 3, 4, 4 ] ) )
+
+		self.assertTrue( testObject.arePrimitiveVariablesValid() )
+
+		pointsScene = GafferScene.ObjectToScene()
+		pointsScene["object"].setValue( testObject )
+
+		deletePoints = GafferScene.DeletePoints()
+
+		deletePoints["in"].setInput( pointsScene["out"] )
+
+		pathFilter = GafferScene.PathFilter( "PathFilter" )
+		pathFilter["paths"].setValue( IECore.StringVectorData( [ '/object' ] ) )
+		deletePoints["filter"].setInput( pathFilter["out"] )
+
+		deletePoints["selectionMode"].setValue( GafferScene.DeletePoints.SelectionMode.IdList )
+		deletePoints["idList"].setValue( IECore.Int64VectorData( [ 1, 7 ] ) )
+
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 10, 12, 13, 14, 15, 16, 18, 19 ] )
+		)
+
+		deletePoints["invert"].setValue( True )
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 11, 17 ] )
+		)
+
+		deletePoints["invert"].setValue( False )
+		deletePoints["selectionMode"].setValue( GafferScene.DeletePoints.SelectionMode.IdListPrimitiveVariable )
+		deletePoints["idListVariable"].setValue( "testA" )
+
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 10, 11, 15, 16, 17 ] )
+		)
+
+		deletePoints["idListVariable"].setValue( "testB" )
+
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 12, 13, 14, 18, 19 ] )
+		)
+
+		deletePoints["invert"].setValue( True )
+
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 10, 11, 15, 16, 17 ] )
+		)
+
+		deletePoints["invert"].setValue( False )
+
+		deletePoints["id"].setValue( "shiftIds" )
+
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 13, 14, 15, 16, 17, 18, 19 ] )
+		)
+
+		# Test that we work with ids outside the range of an Int32
+		deletePoints["id"].setValue( "outOfRangeIds" )
+
+		deletePoints["selectionMode"].setValue( GafferScene.DeletePoints.SelectionMode.IdList )
+		deletePoints["idList"].setValue( IECore.Int64VectorData( [ 8000000001,  8000000002, 8000000007, 8000000008 ] ) )
+
+		self.assertEqual(
+			deletePoints["out"].object( "/object" )["r"].data,
+			IECore.FloatVectorData( [ 10, 13, 14, 15, 16, 19 ] )
+		)
+
+		# With duplicate ids, we only remove the last instance currently. Do we need to do better than this?
+		# Duplicate ids shouldn't exist anyway?
```

Comment:
I think it's worth doing better if it doesn't add complexity or make performance worse. I made some notes on this in the implementation - unless I'm missing something, I don't think it will make anything worse.